tests/surrogate_test.py

```diff
@@ -0,0 +1,206 @@
+                            numpy.array(points)[:, 1],
+                            (-1, 1), (0, 15))
+    branin_at_grids = []
+    for i in transformed_points:
+        branin_at_grids.append(branin(i))
+    return branin_at_grids
+
+
+def test_initialization_1d():
+    """Test if class is properly intiallized."""
+    grid_generator = SmolyakGridGenerator(
+        ChebyshevFirstKind.make_nested_set(2))
+    domain = (-1, 1)
+    surrogate_object = Surrogate(domain, grid_generator)
+    assert numpy.allclose(surrogate_object.domain,
+                          numpy.array(domain, ndmin=2))
+    assert surrogate_object._coefficients is None
+    assert surrogate_object._grid is None
+    grid_object = surrogate_object.grid
+    assert isinstance(grid_object, IndexGrid)
+    assert surrogate_object.dimension == 1
+
+
+def test_initialization_2d():
+    """Test if class is properly intiallized."""
+    grid_generator = SmolyakGridGenerator(
+        ChebyshevFirstKind.make_nested_set(2))
+    domain = [(-5, 5), (-8, 0)]
+    grid_generator_ = 'not an IndexGridGenerator'
+    surrogate_object = Surrogate(domain, grid_generator)
+    assert numpy.allclose(surrogate_object.domain,
+                          numpy.array(domain, ndmin=2))
+    with pytest.raises(TypeError):
+        Surrogate(domain, grid_generator_)
+    assert surrogate_object._coefficients is None
+    grid_object = surrogate_object.grid
+    assert isinstance(grid_object, IndexGrid)
+    assert surrogate_object.dimension == 2
+
+
+def test_map_function_1d():
+    """Test if points are transfromed properly."""
+    grid = SmolyakGridGenerator(ChebyshevFirstKind.make_nested_set(1))
+    domain = (-8, 12)
+    new_domain = (0, 1)
+    f = Surrogate(domain, grid)
+    assert f._mapdomain(2, domain, new_domain) == pytest.approx(0.5)
+    assert numpy.allclose(f._mapdomain([-3, 7],
+                                       domain, new_domain), [0.25, 0.75])
+
+
+def test_map_function_2d():
+    """Test if points are transfromed properly."""
+    grid = SmolyakGridGenerator(ChebyshevFirstKind.make_nested_set(1))
+    domain = ((-10, 10), (0, 2))
+    new_domain = ((0, 1), (-1, 1))
+    f = Surrogate(domain, grid)
+    assert numpy.allclose(f._mapdomain((-10, 0), domain, new_domain), [0, -1])
+    assert numpy.allclose(f._mapdomain([(0, 1), (10, 2)],
+                                       domain, new_domain), [[0.5, 0], [1, 1]])
+
+
+def test_basis_matrix():
+    """Test if a correct basis matrix is generated."""
+    grid_generator = SmolyakGridGenerator(
+        ChebyshevFirstKind.make_nested_set(1))
+    surrogate_object = Surrogate([(-7, 3), (-18, 9)], grid_generator)
+    expected_basis_matrix = numpy.array(
+        [[1, 0, -1, 0, -1], [1, -1, 1, 0, -1], [1, 1, 1, 0, -1],
+         [1, 0, -1, -1, 1], [1, 0, -1, 1, 1]])
+    assert numpy.allclose(expected_basis_matrix,
+                          surrogate_object._make_basis_matrix())
+
```

Comment:
This is an implementation detail that the object is free to change, so don't test it like this. Instead, make sure that training happens correctly elsewhere.